File: src/features/typing/data/java/min_stack.java
import java.util.ArrayList;
import java.util.List;

class MinStack {
    private final List<Integer> stack = new ArrayList<>();
    private final List<Integer> minStack = new ArrayList<>();

    public MinStack() {}

    public void push(int val) {
        stack.add(val);
        int min = val;
        if (!minStack.isEmpty() && minStack.get(minStack.size() - 1) < min) {
            min = minStack.get(minStack.size() - 1);
        }
        minStack.add(min);
    }

    public void pop() {
        stack.remove(stack.size() - 1);
        minStack.remove(minStack.size() - 1);
    }

    public int top() {
        return stack.get(stack.size() - 1);
    }

    public int getMin() {
        return minStack.get(minStack.size() - 1);
    }
}

File: src/features/typing/data/dart/min_stack.dart
class MinStack {
  final List<int> _stack = <int>[];
  final List<int> _minStack = <int>[];

  void push(int val) {
    _stack.add(val);
    var min = val;
    if (_minStack.isNotEmpty && _minStack.last < min) {
      min = _minStack.last;
    }
    _minStack.add(min);
  }

  void pop() {
    _stack.removeLast();
    _minStack.removeLast();
  }

  int top() {
    return _stack.last;
  }

  int getMin() {
    return _minStack.last;
  }
}
